web/tls_config.go

```diff
@@ -297,7 +301,17 @@ func ListenAndServe(server *http.Server, flags *FlagConfig, logger log.Logger) e
 
 	listeners := make([]net.Listener, 0, len(*flags.WebListenAddresses))
 	for _, address := range *flags.WebListenAddresses {
-		listener, err := net.Listen("tcp", address)
+		var err error
+		var listener net.Listener
+		if strings.HasPrefix(address, "vsock://") {
+			var port uint32
+			port, err = parseVsockAddress(address)
```

Comment:
```suggestion
			port, err := parseVsockAddress(address)
```

Reply:
I'm pretty sure it should be ok, it looks like you applied the change.

Reply:
technically , i move the `if err != nil { return err}` into the if scope ( which can use := ) 
but listener still require `=` to make sure it can be used out of if-else scope.

Anyway, current code works as expected.

Reply:
Yes, `listener` still needs a declared var.